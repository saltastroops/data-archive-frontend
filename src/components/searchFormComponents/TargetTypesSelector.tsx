import * as React from "react";
import styled from "styled-components";
import {
  GALAXY,
  ISM,
  SOLAR_SYSTEM_BODY,
  STAR,
  TargetType
} from "../../utils/TargetType";
import { LargeCheckbox } from "../basicComponents/LargeCheckbox";

interface ITargetTypesSelectorProps {
  onChange: (targetTypes: Set<TargetType>) => void;
  targetTypes: Set<TargetType>;
}

const TargetTypesDiv = styled.div`
  && {
    margin-top: 5px;
  }

  label {
    padding-right: 10px;
  }
`;

/**
 * A component for selecting target types.
 *
 * The component does not keep internal state, so it is up to the parent
 * component to ensure that the target types passed as a property remain
 * up-to-date.
 *
 * Properties:
 * -----------
 * targetTypes:
 *     The selected target types.
 * onChange:
 *     The function to call when the user selects or unselects a target type,
 *     which should take the selected target types as its single argument.
 */
export default class TargetTypesSelector extends React.Component<
  ITargetTypesSelectorProps
> {
  onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const updatedTargetTypes = new Set<TargetType>(this.props.targetTypes);
    if (e.target.checked) {
      updatedTargetTypes.add(e.target.value);
    } else {
      updatedTargetTypes.delete(e.target.value);
    }

    this.props.onChange(updatedTargetTypes);
  };

  render() {
    const { targetTypes } = this.props;

    return (
      <TargetTypesDiv>
        <label>
          <LargeCheckbox
            checked={targetTypes.has(GALAXY)}
            data-test="galaxy"
            onChange={this.onChange}
            value={GALAXY}
          />
          Galaxy
        </label>
        <label>
          <LargeCheckbox
            checked={targetTypes.has(ISM)}
            data-test="ism"
            onChange={this.onChange}
            value={ISM}
          />
          ISM
        </label>
        <label>
          <LargeCheckbox
            checked={targetTypes.has(SOLAR_SYSTEM_BODY)}
            data-test="solar-system-body"
            onChange={this.onChange}
            value={SOLAR_SYSTEM_BODY}
          />
          Solar System Body
        </label>
        <label>
          <LargeCheckbox
            checked={targetTypes.has(STAR)}
            data-test="star"
            onChange={this.onChange}
            value={STAR}
          />
          Star
        </label>
      </TargetTypesDiv>
    );
  }
}
