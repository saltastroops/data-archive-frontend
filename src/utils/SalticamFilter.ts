export type SalticamFilter =
  | "SalticamFilter_Name"
  | "340-35"
  | "380-40"
  | "B-S1"
  | "CLR-S1"
  | "Halpha-S1"
  | "Hbn-S1"
  | "Hbw-S1"
  | "I-S1"
  | "R-S1"
  | "Sb-S1"
  | "SDSSg-S1"
  | "SDSSi-S1"
  | "SDSSr-S1"
  | "SDSSu-S1"
  | "SDSSz-S1"
  | "SR613-21"
  | "SR708-25"
  | "SR815-29"
  | "SR862-32"
  | "Su-S1"
  | "Sv-S1"
  | "Sy-S1"
  | "U-S1"
  | "V-S1";

export interface ISalticamFilter {
  name: SalticamFilter;
  descriptiveName: string;
}
