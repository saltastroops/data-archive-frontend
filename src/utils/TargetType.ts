export type TargetType = string;

export const GALAXY = "10";

export const ISM = "20";

export const SOLAR_SYSTEM_BODY = "30";

export const STAR = "40";
