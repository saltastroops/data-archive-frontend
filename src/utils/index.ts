export const toTitleCase = (s: string) => {
  s = s.replace(/_/g, " ");
  return s.replace(/\w\S*/g, (txt: string) => {
    return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
  });
};
