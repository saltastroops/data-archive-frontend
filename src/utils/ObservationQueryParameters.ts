// TypeScript interfaces describing the state of the search form

import { SalticamFilter } from "./SalticamFilter";
import { TargetType } from "./TargetType";

/**
 * An interface describing the React state of the observation query form.
 *
 * This state is used for populating all search form fields (and error messages)
 * and is parsed into the GraphQL query for searching observations.
 *
 * Properties:
 * -----------
 * general:
 *     General information about the observation.
 * target:
 *     Target details.
 * telescope:
 *     Telescope (and instrument) details.
 */
// TODO: Remove loading, replace with IObservationQueryParameters
export interface ISearchFormState {
  general: IGeneral;
  target: ITarget;
  telescope?: ITelescope;
  loading: boolean;
}

/**
 * The available calibration types.
 */
export type CalibrationType = "arc" | "bias" | "flat" | "standard";

/**
 * An interface for query parameters related to general information.
 *
 * Properties:
 * -----------
 * errors:
 *     Error messages.
 * observationNight:
 *     Date of the night when the observation was taken. This
 *     refers to the date whe the night started.
 * proposalCode:
 *     Identifier of the proposal to which the observation belongs.
 * proposalTitle:
 *     Title of the proposal to which the observation belongs.
 * calibrations:
 *     Calibration types to include in the search results
 *
 */
export interface IGeneralErrors {
  observationNight?: string;
  principalInvestigator?: string;
  proposalCode?: string;
  proposalTitle?: string;
  calibrations?: string;
}
export interface IGeneral {
  errors: IGeneralErrors;
  observationNight?: string;
  principalInvestigator?: string;
  proposalCode?: string;
  proposalTitle?: string;
  calibrations: Set<CalibrationType>;
}

/**
 * The available target name resolvers.
 */
export type TargetResolver = "Simbad" | "NED" | "VizieR";

/**
 * The available units for the search cone radius.
 */
export type SearchConeRadiusUnits = "arcseconds" | "arcminutes" | "degrees";

/**
 * An interface for query parameters related to target details.
 *
 * Properties:
 * -----------
 * declination:
 *     Declination.
 * errors:
 *     Errors.
 * name:
 *     Target name.
 * resolver:
 *     Resolver to use for resolving the target name to a position.
 * rightAscension:
 *     Right ascension.
 * searchConeRadius:
 *     Radius of the search cone radius, in the units given by
 *     searchConeRadiusUnits.
 * searchConeRadiusUnits:
 *     Units of the search cone radius given by searchConeRadius.
 *
 */
export interface ITarget {
  declination?: string;
  errors: ITargetErrors;
  loading?: string;
  name?: string;
  resolver: TargetResolver;
  rightAscension?: string;
  searchConeRadius?: string;
  searchConeRadiusUnits: SearchConeRadiusUnits;
  targetTypes: Set<TargetType>;
}

export interface ITargetErrors {
  name?: string;
  declination?: string;
  resolver?: string;
  rightAscension?: string;
  searchConeRadius?: string;
  searchConeRadiusUnits?: string;
  targetTypes?: string;
}

// TELESCOPES

/**
 * The supported telescope names.
 */
export type TelescopeName = "Lesedi" | "SALT" | "1.9 m";

/**
 * An interface for query parameters related to a telescope.
 *
 * This serves as a placeholder for more concrete interfaces such as ISALT.
 *
 * Properties:
 * -----------
 * name:
 *     Telescope name.
 */
export interface ITelescope {
  name?: TelescopeName;
}

/**
 * An interface for query parameters related to SALT.
 *
 * Properties:
 * -----------
 * instrument:
 *     A SALT instrument.
 * name:
 *     The string "SALT".
 */
export interface ISALT extends ITelescope {
  instrument: IInstrument;
  name: "SALT";
}

/**
 * An interface for query parameters related to Lesedi.
 *
 * Properties:
 * -----------
 * instrument:
 *     An instrument used on Lesedi.
 * name:
 *     The string "Lesedi".
 */
export interface ILesedi extends ITelescope {
  instrument: IInstrument;
  name: "Lesedi";
}

/**
 * An interface for query parameters related to the 1.9 m Telescope.
 *
 * Properties:
 * -----------
 * instrument:
 *     An instrument used on the 1./9 m Telescope.
 * name:
 *     The string "1.9 m".
 */
export interface IOneDotNineM extends ITelescope {
  instrument: IInstrument;
  name: "1.9 m";
}

// INSTRUMENTS

/**
 * The available instrument names.
 */
export type InstrumentName =
  | "Salticam"
  | "RSS"
  | "HRS"
  | "BVIT"
  | "SpUpNIC"
  | "SHOC"
  | "HIPPO";

/**
 * An interface for query parameters related to an instrument.
 *
 * This serves as a placeholder for more concrete interfaces such as ISalticam.
 *
 * Properties:
 * -----------
 * name:
 *     Telescope name.
 */
export interface IInstrument {
  name: InstrumentName;
}

/**
 * An interface for query parameters related to Salticam.
 *
 * Properties:
 * -----------
 * detectorMode:
 *     Detector mode.
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "Salticam".
 */
export interface ISalticam extends IInstrument {
  detectorMode?: "Normal" | "Slot Mode";
  errors: {
    detectorMode?: string;
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: SalticamFilter;
  name: "Salticam";
}

/**
 * An interface for query parameters related to RSS.
 *
 * Properties:
 * -----------
 * detectorMode:
 *     Detector mode.
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * mode:
 *     Instrument mode specific details.
 * name:
 *     The string "RSS".
 */
export interface IRSS extends IInstrument {
  detectorMode?: RSSDetectorMode;
  errors: {
    detectorMode?: string;
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  modes?: IRSSModes;
  name: "RSS";
}

export interface IRSSModes {
  errors: {
    fabryPerotMode?: string;
    grating?: string;
    polarimetryModes?: string;
  };
  fabryPerotMode?: RSSFabryPerotMode;
  grating?: RSSGrating;
  names: Set<RSSInstrumentMode>;
  polarimetryModes?: Set<RSSPolarimetryMode>;
}

export type RSSDetectorMode = "Normal" | "Slot Mode" | "";

export type RSSGrating =
  | "Open"
  | "pg0300"
  | "pg0900"
  | "pg1300"
  | "pg1800"
  | "pg2300"
  | "pg3000"
  | "";

export type RSSInstrumentMode =
  | "Fabry Perot"
  | "FP polarimetry"
  | "Imaging"
  | "Polarimetric imaging"
  | "MOS"
  | "MOS polarimetry"
  | "Spectropolarimetry"
  | "Spectroscopy";

export type RSSFabryPerotMode = "HR" | "LR" | "MR" | "TF" | "";

export type RSSPolarimetryMode =
  | "All Stokes"
  | "Circular"
  | "Linear"
  | "Linear Hi"
  | "";

/**
 * An interface for query parameters related to HRS.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * mode:
 *     Resolution mode.
 * name:
 *     The string "HRS".
 */
export interface IHRS extends IInstrument {
  errors: {
    exposureTime?: string;
    mode?: string;
  };
  exposureTime?: string;
  mode?: HRSMode;
  name: "HRS";
}

export type HRSMode =
  | "High Resolution"
  | "High Stability"
  | "Int Cal Fibre"
  | "Low Resolution"
  | "Medium Resolution";

export interface IBVIT extends IInstrument {
  errors: {};
  filter?: BVITFilter;
  mode?: "Imaging" | "Streaming";
  name: "BVIT";
}

export type BVITFilter = "B" | "H-alpha" | "Open" | "R" | "U" | "V";

/**
 * An interface for query parameters related to HIPPO.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "HIPPO".
 */
export interface IHIPPO extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "HIPPO";
}

/**
 * An interface for query parameters related to SHOC.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "SHOC".
 */
export interface ISHOC extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "SHOC";
}

/**
 * An interface for query parameters related to SpUpNIC.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "SpUpNIC".
 */
export interface ISpUpNIC extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "SpUpNIC";
}

export interface IFile {
  id: string;
  filename: string;
  name: string;
  dataType: string;
  isReduced: boolean;
  targetName: string;
  rightAscension: number;
  declination: number;
  observationNight: string;
  category: string;
  telescope: string;
  instrument: string;
  observationId?: string;
  url?: string;
}
export interface IObservation {
  id: string;
  name: string | "Unknown Observation";
  proposal?: string;
  telescope: string;
  startTime: string;
  files: IFile[];
}
