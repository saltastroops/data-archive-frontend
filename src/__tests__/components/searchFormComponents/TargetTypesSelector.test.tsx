import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import wait from "waait";
import TargetTypesSelector from "../../../components/searchFormComponents/TargetTypesSelector";
import { GALAXY, ISM, STAR, TargetType } from "../../../utils/TargetType";

describe("TargetTypesSelector", () => {
  it("should render correctly with no target types", () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <TargetTypesSelector
        onChange={onChange}
        targetTypes={new Set<TargetType>()}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render correctly with target types", () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <TargetTypesSelector
        onChange={onChange}
        targetTypes={new Set<TargetType>([GALAXY, ISM])}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call the onChange method with the correct argument when adding a target type", async () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <TargetTypesSelector
        onChange={onChange}
        targetTypes={new Set<TargetType>([GALAXY, STAR])}
      />
    );
    const ismCheckbox = wrapper.find('input[data-test="ism"]');
    ismCheckbox.simulate("change", { target: { checked: true, value: ISM } });
    await wait(0);
    expect(onChange).toHaveBeenCalledWith(
      new Set<TargetType>([GALAXY, ISM, STAR])
    );
  });

  it("should call the onChange method with the correct argument when removing a target type", async () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <TargetTypesSelector
        onChange={onChange}
        targetTypes={new Set<TargetType>([GALAXY, STAR])}
      />
    );
    const ismCheckbox = wrapper.find('input[data-test="galaxy"]');
    ismCheckbox.simulate("change", {
      target: { checked: false, value: GALAXY }
    });
    await wait(0);
    expect(onChange).toHaveBeenCalledWith(new Set<TargetType>([STAR]));
  });
});
