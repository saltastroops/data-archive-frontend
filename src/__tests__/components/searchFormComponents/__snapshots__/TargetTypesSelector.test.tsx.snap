// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetTypesSelector should render correctly with no target types 1`] = `
.c1.c1 {
  width: 18px;
  height: 18px;
}

.c0.c0 {
  margin-top: 5px;
}

.c0 label {
  padding-right: 10px;
}

<TargetTypesSelector
  onChange={[MockFunction]}
  targetTypes={Set {}}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  && {
    margin-top: 5px;
  }

  label {
    padding-right: 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <label>
          <styled.input
            checked={false}
            data-test="galaxy"
            onChange={[Function]}
            value="10"
          >
            <StyledComponent
              checked={false}
              data-test="galaxy"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="10"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="galaxy"
                onChange={[Function]}
                type="checkbox"
                value="10"
              />
            </StyledComponent>
          </styled.input>
          Galaxy
        </label>
        <label>
          <styled.input
            checked={false}
            data-test="ism"
            onChange={[Function]}
            value="20"
          >
            <StyledComponent
              checked={false}
              data-test="ism"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="20"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="ism"
                onChange={[Function]}
                type="checkbox"
                value="20"
              />
            </StyledComponent>
          </styled.input>
          ISM
        </label>
        <label>
          <styled.input
            checked={false}
            data-test="solar-system-body"
            onChange={[Function]}
            value="30"
          >
            <StyledComponent
              checked={false}
              data-test="solar-system-body"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="30"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="solar-system-body"
                onChange={[Function]}
                type="checkbox"
                value="30"
              />
            </StyledComponent>
          </styled.input>
          Solar System Body
        </label>
        <label>
          <styled.input
            checked={false}
            data-test="star"
            onChange={[Function]}
            value="40"
          >
            <StyledComponent
              checked={false}
              data-test="star"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="40"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="star"
                onChange={[Function]}
                type="checkbox"
                value="40"
              />
            </StyledComponent>
          </styled.input>
          Star
        </label>
      </div>
    </StyledComponent>
  </styled.div>
</TargetTypesSelector>
`;

exports[`TargetTypesSelector should render correctly with target types 1`] = `
.c1.c1 {
  width: 18px;
  height: 18px;
}

.c0.c0 {
  margin-top: 5px;
}

.c0 label {
  padding-right: 10px;
}

<TargetTypesSelector
  onChange={[MockFunction]}
  targetTypes={
    Set {
      "10",
      "20",
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  && {
    margin-top: 5px;
  }

  label {
    padding-right: 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <label>
          <styled.input
            checked={true}
            data-test="galaxy"
            onChange={[Function]}
            value="10"
          >
            <StyledComponent
              checked={true}
              data-test="galaxy"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="10"
            >
              <input
                checked={true}
                className="checkbox c1"
                data-test="galaxy"
                onChange={[Function]}
                type="checkbox"
                value="10"
              />
            </StyledComponent>
          </styled.input>
          Galaxy
        </label>
        <label>
          <styled.input
            checked={true}
            data-test="ism"
            onChange={[Function]}
            value="20"
          >
            <StyledComponent
              checked={true}
              data-test="ism"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="20"
            >
              <input
                checked={true}
                className="checkbox c1"
                data-test="ism"
                onChange={[Function]}
                type="checkbox"
                value="20"
              />
            </StyledComponent>
          </styled.input>
          ISM
        </label>
        <label>
          <styled.input
            checked={false}
            data-test="solar-system-body"
            onChange={[Function]}
            value="30"
          >
            <StyledComponent
              checked={false}
              data-test="solar-system-body"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="30"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="solar-system-body"
                onChange={[Function]}
                type="checkbox"
                value="30"
              />
            </StyledComponent>
          </styled.input>
          Solar System Body
        </label>
        <label>
          <styled.input
            checked={false}
            data-test="star"
            onChange={[Function]}
            value="40"
          >
            <StyledComponent
              checked={false}
              data-test="star"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="40"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="star"
                onChange={[Function]}
                type="checkbox"
                value="40"
              />
            </StyledComponent>
          </styled.input>
          Star
        </label>
      </div>
    </StyledComponent>
  </styled.div>
</TargetTypesSelector>
`;
